data/scenarios/Challenges/Ranching/beekeeping.yaml

```diff
@@ -0,0 +1,453 @@
+version: 1
+name: Beekeeping
+author: Karl Ostmo
+description: |
+  Bootstrap an organic *bee*verage industry.
+creative: false
+seed: 2
+attrs:
+  - name: bee
+    fg: '#ffff00'
+    bg: '#000000'
+  - name: water_cask
+    fg: '#4488ff'
+    bg: '#8B4513'
+  - name: mybase
+    fg: '#e0c0e0'
+  - name: iceblue
+    fg: '#ddddff'
+objectives:
+  - teaser: Apiarist
+    goal:
+      - |
+        Build a `beehive`{=structure}.
+      - |
+        This will attract bees, which will gather
+        `nectar`{=entity} from nearby `wildflower`{=entity}s.
+      - |
+        Do not crowd hives; provide at least one cell margin
+        between them.
+      - |
+        Given that `wildflower`{=entity}s cannot be relocated,
+        judicious placement is essential for efficient
+        `honeycomb`{=entity} production.
+    condition: |
+      foundStructure <- structure "beehive" 0;
+      return $ case foundStructure (\_. false) (\_. true);
+  - teaser: Collect honeycomb
+    goal:
+      - |
+        Collect `honeycomb`{=entity} from the `beehive`{=structure}s.
+      - |
+        After gathering a certain amount of nectar, a bee will return
+        to its hive and place honeycomb alongside it.
+        A finite amount of honeycomb may be accumulated before the
+        bee becomes dormant.
+    condition: |
+      as base {
+        has "honeycomb"
+      }
+  - teaser: Cooper
+    goal:
+      - |
+        Make a `cask`{=entity}.
+      - |
+        Utilize local timber and your on-hand supply of iron material.
+      - |
+        You may `use "siphon" forward` when positioned in front of
+        a lake to fill a `cask`{=entity} with `lakewater`{=entity}.
+    condition: |
+      as base {
+        has "cask"
+      }
+  - teaser: Brewmeister
+    goal:
+      - |
+        Ferment 2 barrels of `mead`{=entity}.
+    condition: |
+      as base {
+        meadCount <- count "mead";
+        return $ meadCount >= 2;
+      }
+  - teaser: Tavern keeper
+    optional: true
+    goal:
+      - |
+        Construct a `mead hall`{=structure}.
+    condition: |
+      foundStructure <- structure "mead hall" 0;
+      return $ case foundStructure (\_. false) (\_. true);
+robots:
+  - name: base
+    display:
+      attr: mybase
+    dir: [1, 0]
+    devices:
+      - 3D printer
+      - blueprint
```

Comment:
I didn't use them, but now I see that it would be nice to be able to program a robot to go around and collect honeycombs by iterating through beehive structures.

Apropos of nothing, is there a distance limit on how far away you can detect structures with the `structure` command?

Reply:
> is there a distance limit on how far away you can detect structures with the `structure` command?

No limit, and I suppose this does facilitate instantaneous communication.  Let's just say `blueprint` comes with an internet [MLS](https://en.wikipedia.org/wiki/Multiple_listing_service) subscription.

Reply:
OK, makes sense, we'll just have to be careful if we ever decide to introduce the `blueprint` device into classic mode.